Add Apple Pay and Google Pay to payment method list in gateway description

anu-rock opened the pull request:
#### Changes proposed in this Pull Request

* Match the descriptions of both gateways (WCPay and UPE) to the text on the official website.
* The revised text includes Apple Pay and Google Pay.

#### Testing instructions

* Go to WP Admin → WooCommerce → Settings → Payments → All payment methods.
* Check the description for WooCommerce Payments.
* The description should be the same with UPE enabled and disabled.

Before, with UPE disabled:

<img width="1355" alt="image" src="https://user-images.githubusercontent.com/1288616/212042663-9d5d9b62-39c9-48d7-bc34-eece339e2ebc.png">

Before, with UPE enabled:

<img width="1357" alt="image" src="https://user-images.githubusercontent.com/1288616/212042812-9243bcd6-0906-4fa7-886a-15cdb3977dfa.png">

After, with UPE disabled:

<img width="1359" alt="image" src="https://user-images.githubusercontent.com/1288616/212045790-84d7d286-5f94-4820-a0d2-5e74056e609b.png">

After, with UPE enabled:

<img width="1356" alt="image" src="https://user-images.githubusercontent.com/1288616/212042513-e068f5af-8f1c-4791-a674-de5a479c197a.png">


#### Additional context

paJDYF-5Wk-p2


-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (does not apply)
- [ ] Tested on mobile (does not apply)

**Post merge**

<!--
Make sure you edit the page for the current release when adding testing instructions.
We often create a blank page ahead of time for the next release.
If this PR need not be QA tested, edit to 'QA Testing Not Applicable'
-->

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

anu-rock committed:
Match the descriptions of both gateways to text on website

anu-rock committed:
Add changelog

kalessil reviewed (changes requested):
The description differs, probably we shouldn't override `title`, `method_description`, `method_title` and `description` in `\WCPay\Payment_Methods\UPE_Payment_Gateway::__construct` at all. Those will be initiated in `\WC_Payment_Gateway_WCPay::__construct`.

With UPE enabled:
![Screenshot from 2023-01-12 10-26-10](https://user-images.githubusercontent.com/1577185/212030139-c01543ed-8b33-42f8-bea0-e8c2f99f9595.png)

With UPE disabled:
![Screenshot from 2023-01-12 10-31-53](https://user-images.githubusercontent.com/1577185/212030354-e0561c9a-78bc-436f-aa39-4b4e4238a200.png)

anu-rock committed:
Remove title and desc overrides in UPE gateway

title, description, method_title, and method_description fields in
UPE gateway were similar to ones in WCPay. So, we'll let them be inherited.

anu-rock committed:
Fix gateway title in UPE

anu-rock commented:
@kalessil Thanks for the tip! That does make sense. I've pushed new commits to address that. I've also updated the testing instructions.

I learned that we need to keep the `title` override in UPE. Otherwise, the method `maybe_filter_gateway_title` gets broken along with its tests.

Also noting that the "Credit card / debit card" title that we see with UPE disabled probably also needs to be updated as it feels out of place on the all payment methods page. Modifying it may be beyond the scope of this PR since it'll impact the title we see on the checkout page too.

kalessil committed:
UPE: tweak gateway properties population.

kalessil reviewed (approved):
LGTM! Asked for confirmation from product folks: paJDYF-5Wk-p2#comment-15661

kalessil committed:
Merge branch 'develop' into update/gateway-descriptions-add-apple-google-pay
